blocks/algorithms/__init__.py

```diff
@@ -575,6 +577,64 @@ def compute_steps(self, previous_steps):
         return steps, []
 
 
+class VariableClipping(StepRule):
+    """Clip the maximum norm of individual variables along certain axes.
+
+    This :class:`StepRule` can be used to implement L2 norm constraints on
+    individual hidden units, groups of hidden units, convolutional
```

Comment:
If you had convolutional filters of the form `(n_in, n_out, filter_height, filter_width)`, for example, passing axes=(2, 3) would restrict the norm of each `(filter_height, filter_width)` element.